packages/eventindexer/indexer/fetch_nft_metadata.go

```diff
@@ -0,0 +1,167 @@
+package indexer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/taikoxyz/taiko-mono/packages/eventindexer"
+	"github.com/taikoxyz/taiko-mono/packages/eventindexer/contracts/erc1155"
+	"github.com/taikoxyz/taiko-mono/packages/eventindexer/contracts/erc721"
+)
+
+func (i *Indexer) fetchNFTMetadata(
+	ctx context.Context, contractAddress string,
+	tokenID *big.Int,
+	abiJSON string,
+	methodName string,
+	chainID *big.Int) (*eventindexer.NFTMetadata, error) {
+	contractABI, err := abi.JSON(strings.NewReader(abiJSON))
+	if err != nil {
+		return nil, err
+	}
+
+	contractAddressCommon := common.HexToAddress(contractAddress)
+
+	callData, err := contractABI.Pack(methodName, tokenID)
+	if err != nil {
+		return nil, err
+	}
+
+	msg := ethereum.CallMsg{
+		To:   &contractAddressCommon,
+		Data: callData,
+	}
+
+	result, err := i.ethClient.CallContract(ctx, msg, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var tokenURI string
+
+	err = contractABI.UnpackIntoInterface(&tokenURI, methodName, result)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := resolveMetadataURL(tokenURI)
+	if err != nil {
+		if errors.Is(err, eventindexer.ErrInvalidURL) {
+			slog.Warn("Skipping metadata due to invalid URI")
```

Comment:
can we update this log? `slog.Warn("Skipping metadata due to invalid URI", "tokenID", tokenID.String(), "contractAddress", contractAddress)`.. Would help us look at some that encounter this.